client/service/src/builder.rs

```diff
@@ -62,6 +62,11 @@ struct ServiceMetrics {
 	cpu_usage_percentage: Gauge<F64>,
 	network_per_sec_bytes: GaugeVec<U64>,
 	node_roles: Gauge<U64>,
+	database_cache: Gauge<U64>,
+	state_cache: Gauge<U64>,
+	state_db_non_canonical: Gauge<U64>,
+	state_db_pruning: Gauge<U64>,
+	state_db_pinned: Gauge<U64>,
```

Comment:
I think it would be better to turn these 3 into a `state_db_cache_bytes` `GaugeVec`, with a label `type` or something.